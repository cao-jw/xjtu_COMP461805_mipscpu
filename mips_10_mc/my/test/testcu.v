`include "CU.v"
`include "SR.v"
module test(CLK,Op);
    input CLK;
    input [5:0] Op;
    wire PCWr,PCWrCond,IorD,MemRd,MemWr,IRWr,MemtoReg,ALUSrcA,RegWr,RegDst;
    wire[1:0] PCSrc,ALUOp,ALUSrcB;
    wire [3:0] S;
    wire [3:0] NS;
    SR sr(.S(S),.NS(NS),.CLK(CLK));
    CU cu(.PCWr(PCWr),
    .PCWrCond(PCWrCond),
    .IorD(IorD),
    .MemRd(MemRd),
    .MemWr(MemWr),
    .IRWr(IRWr),
    .MemtoReg(MemtoReg),
    .ALUSrcA(ALUSrcA),
    .RegWr(RegWr),
    .RegDst(RegDst),
    .PCSrc(PCSrc),
    .ALUOp(ALUOp),
    .ALUSrcB(ALUSrcB),
    .S(S),
    .NS(NS),
    .Op(Op));
endmodule
module testbench;
reg clk;
reg [5:0]Op;
initial begin
    clk=0;
    Op=6'b100011;
    $dumpfile("CU.vcd");
    $dumpvars;
    #200;
    $finish;
end
always begin
    #10;
    clk=~clk;
end
test w(.CLK(clk),.Op(Op));
endmodule