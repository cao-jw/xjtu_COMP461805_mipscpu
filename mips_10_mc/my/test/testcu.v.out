#! /usr/local/iverilog/bin/vvp
:ivl_version "11.0 (devel)" "(s20150603-642-g3bdb50da)";
:ivl_delay_selection "TYPICAL";
:vpi_time_precision + 0;
:vpi_module "system";
:vpi_module "vhdl_sys";
:vpi_module "vhdl_textio";
:vpi_module "v2005_math";
:vpi_module "va_math";
S_00000000011d6460 .scope module, "testbench" "testbench" 2 28;
 .timescale 0 0;
v000000000122c9e0_0 .var "Op", 5 0;
v000000000122c620_0 .var "clk", 0 0;
S_00000000011d3bd0 .scope module, "w" "test" 2 43, 2 3 0, S_00000000011d6460;
 .timescale 0 0;
    .port_info 0 /INPUT 1 "CLK";
    .port_info 1 /INPUT 6 "Op";
v000000000122cf80_0 .net "ALUOp", 1 0, v00000000011d3ef0_0;  1 drivers
v000000000122c300_0 .net "ALUSrcA", 0 0, v00000000011d65f0_0;  1 drivers
v000000000122cbc0_0 .net "ALUSrcB", 1 0, v00000000011d6690_0;  1 drivers
v000000000122cd00_0 .net "CLK", 0 0, v000000000122c620_0;  1 drivers
v000000000122ce40_0 .net "IRWr", 0 0, v0000000001176be0_0;  1 drivers
v000000000122cee0_0 .net "IorD", 0 0, v0000000001176c80_0;  1 drivers
v000000000122c120_0 .net "MemRd", 0 0, v0000000001176d20_0;  1 drivers
v000000000122c260_0 .net "MemWr", 0 0, v0000000001176dc0_0;  1 drivers
v000000000122c3a0_0 .net "MemtoReg", 0 0, v0000000001176e60_0;  1 drivers
v000000000122c760_0 .net "NS", 3 0, v0000000001176f00_0;  1 drivers
v000000000122c440_0 .net "Op", 5 0, v000000000122c9e0_0;  1 drivers
v000000000122c4e0_0 .net "PCSrc", 1 0, v000000000117b470_0;  1 drivers
v000000000122c800_0 .net "PCWr", 0 0, v000000000117b510_0;  1 drivers
v000000000122c580_0 .net "PCWrCond", 0 0, v000000000117b5b0_0;  1 drivers
v000000000122c8a0_0 .net "RegDst", 0 0, v000000000117b650_0;  1 drivers
v000000000122cda0_0 .net "RegWr", 0 0, v000000000122cc60_0;  1 drivers
v000000000122c940_0 .net "S", 3 0, v000000000122cb20_0;  1 drivers
S_00000000011d3d60 .scope module, "cu" "CU" 2 11, 3 1 0, S_00000000011d3bd0;
 .timescale 0 0;
    .port_info 0 /INPUT 6 "Op";
    .port_info 1 /INPUT 4 "S";
    .port_info 2 /OUTPUT 1 "PCWr";
    .port_info 3 /OUTPUT 1 "PCWrCond";
    .port_info 4 /OUTPUT 1 "IorD";
    .port_info 5 /OUTPUT 1 "MemRd";
    .port_info 6 /OUTPUT 1 "MemWr";
    .port_info 7 /OUTPUT 1 "IRWr";
    .port_info 8 /OUTPUT 1 "MemtoReg";
    .port_info 9 /OUTPUT 2 "PCSrc";
    .port_info 10 /OUTPUT 2 "ALUOp";
    .port_info 11 /OUTPUT 2 "ALUSrcB";
    .port_info 12 /OUTPUT 1 "ALUSrcA";
    .port_info 13 /NODIR 0 "";
    .port_info 14 /OUTPUT 1 "RegWr";
    .port_info 15 /OUTPUT 1 "RegDst";
    .port_info 16 /OUTPUT 4 "NS";
v00000000011d3ef0_0 .var "ALUOp", 1 0;
v00000000011d65f0_0 .var "ALUSrcA", 0 0;
v00000000011d6690_0 .var "ALUSrcB", 1 0;
v0000000001176be0_0 .var "IRWr", 0 0;
v0000000001176c80_0 .var "IorD", 0 0;
v0000000001176d20_0 .var "MemRd", 0 0;
v0000000001176dc0_0 .var "MemWr", 0 0;
v0000000001176e60_0 .var "MemtoReg", 0 0;
v0000000001176f00_0 .var "NS", 3 0;
v000000000117b3d0_0 .net "Op", 5 0, v000000000122c9e0_0;  alias, 1 drivers
v000000000117b470_0 .var "PCSrc", 1 0;
v000000000117b510_0 .var "PCWr", 0 0;
v000000000117b5b0_0 .var "PCWrCond", 0 0;
v000000000117b650_0 .var "RegDst", 0 0;
v000000000122cc60_0 .var "RegWr", 0 0;
v000000000122c080_0 .net "S", 3 0, v000000000122cb20_0;  alias, 1 drivers
E_00000000011b73b0 .event edge, v000000000122c080_0, v000000000117b3d0_0;
S_00000000011c2fb0 .scope module, "sr" "SR" 2 10, 4 1 0, S_00000000011d3bd0;
 .timescale 0 0;
    .port_info 0 /INPUT 1 "CLK";
    .port_info 1 /OUTPUT 4 "S";
    .port_info 2 /INPUT 4 "NS";
v000000000122c1c0_0 .net "CLK", 0 0, v000000000122c620_0;  alias, 1 drivers
v000000000122c6c0_0 .net "NS", 3 0, v0000000001176f00_0;  alias, 1 drivers
v000000000122cb20_0 .var "S", 3 0;
E_00000000011b7670 .event posedge, v000000000122c1c0_0;
    .scope S_00000000011c2fb0;
T_0 ;
    %wait E_00000000011b7670;
    %load/vec4 v000000000122c6c0_0;
    %store/vec4 v000000000122cb20_0, 0, 4;
    %jmp T_0;
    .thread T_0;
    .scope S_00000000011c2fb0;
T_1 ;
    %pushi/vec4 0, 0, 4;
    %store/vec4 v000000000122cb20_0, 0, 4;
    %end;
    .thread T_1;
    .scope S_00000000011d3d60;
T_2 ;
    %wait E_00000000011b73b0;
    %load/vec4 v000000000122c080_0;
    %dup/vec4;
    %pushi/vec4 0, 0, 4;
    %cmp/u;
    %jmp/1 T_2.0, 6;
    %dup/vec4;
    %pushi/vec4 1, 0, 4;
    %cmp/u;
    %jmp/1 T_2.1, 6;
    %dup/vec4;
    %pushi/vec4 2, 0, 4;
    %cmp/u;
    %jmp/1 T_2.2, 6;
    %dup/vec4;
    %pushi/vec4 3, 0, 4;
    %cmp/u;
    %jmp/1 T_2.3, 6;
    %dup/vec4;
    %pushi/vec4 4, 0, 4;
    %cmp/u;
    %jmp/1 T_2.4, 6;
    %dup/vec4;
    %pushi/vec4 5, 0, 4;
    %cmp/u;
    %jmp/1 T_2.5, 6;
    %dup/vec4;
    %pushi/vec4 6, 0, 4;
    %cmp/u;
    %jmp/1 T_2.6, 6;
    %dup/vec4;
    %pushi/vec4 7, 0, 4;
    %cmp/u;
    %jmp/1 T_2.7, 6;
    %dup/vec4;
    %pushi/vec4 8, 0, 4;
    %cmp/u;
    %jmp/1 T_2.8, 6;
    %dup/vec4;
    %pushi/vec4 9, 0, 4;
    %cmp/u;
    %jmp/1 T_2.9, 6;
    %jmp T_2.10;
T_2.0 ;
    %pushi/vec4 37896, 0, 16;
    %split/vec4 1;
    %store/vec4 v000000000117b650_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v000000000122cc60_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v00000000011d65f0_0, 0, 1;
    %split/vec4 2;
    %store/vec4 v00000000011d6690_0, 0, 2;
    %split/vec4 2;
    %store/vec4 v00000000011d3ef0_0, 0, 2;
    %split/vec4 2;
    %store/vec4 v000000000117b470_0, 0, 2;
    %split/vec4 1;
    %store/vec4 v0000000001176e60_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v0000000001176be0_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v0000000001176dc0_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v0000000001176d20_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v0000000001176c80_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v000000000117b5b0_0, 0, 1;
    %store/vec4 v000000000117b510_0, 0, 1;
    %pushi/vec4 1, 0, 4;
    %store/vec4 v0000000001176f00_0, 0, 4;
    %jmp T_2.10;
T_2.1 ;
    %pushi/vec4 24, 0, 16;
    %split/vec4 1;
    %store/vec4 v000000000117b650_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v000000000122cc60_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v00000000011d65f0_0, 0, 1;
    %split/vec4 2;
    %store/vec4 v00000000011d6690_0, 0, 2;
    %split/vec4 2;
    %store/vec4 v00000000011d3ef0_0, 0, 2;
    %split/vec4 2;
    %store/vec4 v000000000117b470_0, 0, 2;
    %split/vec4 1;
    %store/vec4 v0000000001176e60_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v0000000001176be0_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v0000000001176dc0_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v0000000001176d20_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v0000000001176c80_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v000000000117b5b0_0, 0, 1;
    %store/vec4 v000000000117b510_0, 0, 1;
    %load/vec4 v000000000117b3d0_0;
    %dup/vec4;
    %pushi/vec4 2, 0, 6;
    %cmp/u;
    %jmp/1 T_2.11, 6;
    %dup/vec4;
    %pushi/vec4 4, 0, 6;
    %cmp/u;
    %jmp/1 T_2.12, 6;
    %dup/vec4;
    %pushi/vec4 0, 0, 6;
    %cmp/u;
    %jmp/1 T_2.13, 6;
    %dup/vec4;
    %pushi/vec4 43, 0, 6;
    %cmp/u;
    %jmp/1 T_2.14, 6;
    %dup/vec4;
    %pushi/vec4 35, 0, 6;
    %cmp/u;
    %jmp/1 T_2.15, 6;
    %jmp T_2.16;
T_2.11 ;
    %pushi/vec4 9, 0, 4;
    %store/vec4 v0000000001176f00_0, 0, 4;
    %jmp T_2.16;
T_2.12 ;
    %pushi/vec4 8, 0, 4;
    %store/vec4 v0000000001176f00_0, 0, 4;
    %jmp T_2.16;
T_2.13 ;
    %pushi/vec4 6, 0, 4;
    %store/vec4 v0000000001176f00_0, 0, 4;
    %jmp T_2.16;
T_2.14 ;
    %pushi/vec4 2, 0, 4;
    %store/vec4 v0000000001176f00_0, 0, 4;
    %jmp T_2.16;
T_2.15 ;
    %pushi/vec4 2, 0, 4;
    %store/vec4 v0000000001176f00_0, 0, 4;
    %jmp T_2.16;
T_2.16 ;
    %pop/vec4 1;
    %jmp T_2.10;
T_2.2 ;
    %pushi/vec4 20, 0, 16;
    %split/vec4 1;
    %store/vec4 v000000000117b650_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v000000000122cc60_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v00000000011d65f0_0, 0, 1;
    %split/vec4 2;
    %store/vec4 v00000000011d6690_0, 0, 2;
    %split/vec4 2;
    %store/vec4 v00000000011d3ef0_0, 0, 2;
    %split/vec4 2;
    %store/vec4 v000000000117b470_0, 0, 2;
    %split/vec4 1;
    %store/vec4 v0000000001176e60_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v0000000001176be0_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v0000000001176dc0_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v0000000001176d20_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v0000000001176c80_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v000000000117b5b0_0, 0, 1;
    %store/vec4 v000000000117b510_0, 0, 1;
    %load/vec4 v000000000117b3d0_0;
    %dup/vec4;
    %pushi/vec4 35, 0, 6;
    %cmp/u;
    %jmp/1 T_2.17, 6;
    %dup/vec4;
    %pushi/vec4 43, 0, 6;
    %cmp/u;
    %jmp/1 T_2.18, 6;
    %jmp T_2.19;
T_2.17 ;
    %pushi/vec4 3, 0, 4;
    %store/vec4 v0000000001176f00_0, 0, 4;
    %jmp T_2.19;
T_2.18 ;
    %pushi/vec4 5, 0, 4;
    %store/vec4 v0000000001176f00_0, 0, 4;
    %jmp T_2.19;
T_2.19 ;
    %pop/vec4 1;
    %jmp T_2.10;
T_2.3 ;
    %pushi/vec4 12288, 0, 16;
    %split/vec4 1;
    %store/vec4 v000000000117b650_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v000000000122cc60_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v00000000011d65f0_0, 0, 1;
    %split/vec4 2;
    %store/vec4 v00000000011d6690_0, 0, 2;
    %split/vec4 2;
    %store/vec4 v00000000011d3ef0_0, 0, 2;
    %split/vec4 2;
    %store/vec4 v000000000117b470_0, 0, 2;
    %split/vec4 1;
    %store/vec4 v0000000001176e60_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v0000000001176be0_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v0000000001176dc0_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v0000000001176d20_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v0000000001176c80_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v000000000117b5b0_0, 0, 1;
    %store/vec4 v000000000117b510_0, 0, 1;
    %pushi/vec4 4, 0, 4;
    %store/vec4 v0000000001176f00_0, 0, 4;
    %jmp T_2.10;
T_2.4 ;
    %pushi/vec4 514, 0, 16;
    %split/vec4 1;
    %store/vec4 v000000000117b650_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v000000000122cc60_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v00000000011d65f0_0, 0, 1;
    %split/vec4 2;
    %store/vec4 v00000000011d6690_0, 0, 2;
    %split/vec4 2;
    %store/vec4 v00000000011d3ef0_0, 0, 2;
    %split/vec4 2;
    %store/vec4 v000000000117b470_0, 0, 2;
    %split/vec4 1;
    %store/vec4 v0000000001176e60_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v0000000001176be0_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v0000000001176dc0_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v0000000001176d20_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v0000000001176c80_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v000000000117b5b0_0, 0, 1;
    %store/vec4 v000000000117b510_0, 0, 1;
    %pushi/vec4 0, 0, 4;
    %store/vec4 v0000000001176f00_0, 0, 4;
    %jmp T_2.10;
T_2.5 ;
    %pushi/vec4 10240, 0, 16;
    %split/vec4 1;
    %store/vec4 v000000000117b650_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v000000000122cc60_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v00000000011d65f0_0, 0, 1;
    %split/vec4 2;
    %store/vec4 v00000000011d6690_0, 0, 2;
    %split/vec4 2;
    %store/vec4 v00000000011d3ef0_0, 0, 2;
    %split/vec4 2;
    %store/vec4 v000000000117b470_0, 0, 2;
    %split/vec4 1;
    %store/vec4 v0000000001176e60_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v0000000001176be0_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v0000000001176dc0_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v0000000001176d20_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v0000000001176c80_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v000000000117b5b0_0, 0, 1;
    %store/vec4 v000000000117b510_0, 0, 1;
    %pushi/vec4 0, 0, 4;
    %store/vec4 v0000000001176f00_0, 0, 4;
    %jmp T_2.10;
T_2.6 ;
    %pushi/vec4 68, 0, 16;
    %split/vec4 1;
    %store/vec4 v000000000117b650_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v000000000122cc60_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v00000000011d65f0_0, 0, 1;
    %split/vec4 2;
    %store/vec4 v00000000011d6690_0, 0, 2;
    %split/vec4 2;
    %store/vec4 v00000000011d3ef0_0, 0, 2;
    %split/vec4 2;
    %store/vec4 v000000000117b470_0, 0, 2;
    %split/vec4 1;
    %store/vec4 v0000000001176e60_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v0000000001176be0_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v0000000001176dc0_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v0000000001176d20_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v0000000001176c80_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v000000000117b5b0_0, 0, 1;
    %store/vec4 v000000000117b510_0, 0, 1;
    %pushi/vec4 7, 0, 4;
    %store/vec4 v0000000001176f00_0, 0, 4;
    %jmp T_2.10;
T_2.7 ;
    %pushi/vec4 3, 0, 16;
    %split/vec4 1;
    %store/vec4 v000000000117b650_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v000000000122cc60_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v00000000011d65f0_0, 0, 1;
    %split/vec4 2;
    %store/vec4 v00000000011d6690_0, 0, 2;
    %split/vec4 2;
    %store/vec4 v00000000011d3ef0_0, 0, 2;
    %split/vec4 2;
    %store/vec4 v000000000117b470_0, 0, 2;
    %split/vec4 1;
    %store/vec4 v0000000001176e60_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v0000000001176be0_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v0000000001176dc0_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v0000000001176d20_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v0000000001176c80_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v000000000117b5b0_0, 0, 1;
    %store/vec4 v000000000117b510_0, 0, 1;
    %pushi/vec4 0, 0, 4;
    %store/vec4 v0000000001176f00_0, 0, 4;
    %jmp T_2.10;
T_2.8 ;
    %pushi/vec4 16548, 0, 16;
    %split/vec4 1;
    %store/vec4 v000000000117b650_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v000000000122cc60_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v00000000011d65f0_0, 0, 1;
    %split/vec4 2;
    %store/vec4 v00000000011d6690_0, 0, 2;
    %split/vec4 2;
    %store/vec4 v00000000011d3ef0_0, 0, 2;
    %split/vec4 2;
    %store/vec4 v000000000117b470_0, 0, 2;
    %split/vec4 1;
    %store/vec4 v0000000001176e60_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v0000000001176be0_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v0000000001176dc0_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v0000000001176d20_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v0000000001176c80_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v000000000117b5b0_0, 0, 1;
    %store/vec4 v000000000117b510_0, 0, 1;
    %pushi/vec4 0, 0, 4;
    %store/vec4 v0000000001176f00_0, 0, 4;
    %jmp T_2.10;
T_2.9 ;
    %pushi/vec4 33024, 0, 16;
    %split/vec4 1;
    %store/vec4 v000000000117b650_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v000000000122cc60_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v00000000011d65f0_0, 0, 1;
    %split/vec4 2;
    %store/vec4 v00000000011d6690_0, 0, 2;
    %split/vec4 2;
    %store/vec4 v00000000011d3ef0_0, 0, 2;
    %split/vec4 2;
    %store/vec4 v000000000117b470_0, 0, 2;
    %split/vec4 1;
    %store/vec4 v0000000001176e60_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v0000000001176be0_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v0000000001176dc0_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v0000000001176d20_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v0000000001176c80_0, 0, 1;
    %split/vec4 1;
    %store/vec4 v000000000117b5b0_0, 0, 1;
    %store/vec4 v000000000117b510_0, 0, 1;
    %pushi/vec4 0, 0, 4;
    %store/vec4 v0000000001176f00_0, 0, 4;
    %jmp T_2.10;
T_2.10 ;
    %pop/vec4 1;
    %jmp T_2;
    .thread T_2, $push;
    .scope S_00000000011d6460;
T_3 ;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v000000000122c620_0, 0, 1;
    %pushi/vec4 35, 0, 6;
    %store/vec4 v000000000122c9e0_0, 0, 6;
    %vpi_call 2 34 "$dumpfile", "CU.vcd" {0 0 0};
    %vpi_call 2 35 "$dumpvars" {0 0 0};
    %delay 200, 0;
    %vpi_call 2 37 "$finish" {0 0 0};
    %end;
    .thread T_3;
    .scope S_00000000011d6460;
T_4 ;
    %delay 10, 0;
    %load/vec4 v000000000122c620_0;
    %inv;
    %store/vec4 v000000000122c620_0, 0, 1;
    %jmp T_4;
    .thread T_4;
# The file index is used to find the file name in the following table.
:file_names 5;
    "N/A";
    "<interactive>";
    "testcu.v";
    "./CU.v";
    "./SR.v";
